sdk/src/main/java/io/opentelemetry/sdk/common/ConfigBuilder.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.common;
+
+import java.util.Map;
+
+public abstract class ConfigBuilder {
+
+  protected ConfigBuilder() {}
+
+  public abstract ConfigBuilder fromConfigMap(Map<String, String> configMap);
```

Comment:
What naming conventions do we expect in configMap? `ENV_VAR_CONVENTIONS`? Or `systemProperty.conventions`? It seems the former.

Reply:
From https://github.com/open-telemetry/opentelemetry-java/issues/996#issuecomment-597374377 it is the former, yes.
@bogdandrutu and @trask could you please provide more input on this?

Reply:
I like the convention of looking for system properties with `one.twoThree` and looking for the same env var with `ONE_TWO_THREE`. Since the translation is not reversible (does `ONE_TWO_THREE` map to `one.two.three` or `oneTwo.three` or ...), I think it makes sense to use the system property names in the internal map.